src/app/project/documents/documents-tab.component.ts

```diff
@@ -104,6 +123,99 @@ export class DocumentsTabComponent implements OnInit, OnDestroy {
       );
   }
 
+  /**
+   * Groups files together within their sections, then sorts the sections.
+   */
+  groupAndSortFilesWithinSections(): void {
+    // Get a map of section IDs to their names.
+    const sectionIdsToNames = mapValues(keyBy(this.documentSections, '_id'), 'name');
+
+    /*
+    * If there are document/file sections, partition the list of
+    * files into those that are sectioned and those that aren't.
+    * */
+    const [filesWithSections, unorganizedFiles] = partition(
+      this.documents,
+      (document) => document.section && Object.keys(sectionIdsToNames).includes(document.section)
+    );
+
+    // Set documents in the "main" table to be docs without sections.
+    this.documents = unorganizedFiles;
+    // Update "main" table docs count.
+    this.tableParams.totalListItems = unorganizedFiles.length;
+
+    // Have each file use the section name rather than ID.
+    filesWithSections.forEach(file => {
+      file.section = sectionIdsToNames[file.section];
+    })
+
+    // Group files together with by section.
+    const unorderedDocumentGroupings = groupBy(filesWithSections, 'section');
+
+    // Convert grouped files object to array to ensure correction section order is used.
```

Comment:
```suggestion
    // Convert grouped files object to array to ensure correct section order is used.
```

Reply:
I guess I skipped the edit stage on this doc block. Thanks!